caliban/resources/caliban_wrapper.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''caliban wrapper utility'''
+
+import argparse
+import copy
+import json
+import logging
+import os
+import subprocess
+import sys
+import time
+
+logging.basicConfig(level=logging.INFO)
+
+
+# ----------------------------------------------------------------------------
+def _parse_kv_pair(s):
+  """
+    Parse a key, value pair, separated by '='
+
+    On the command line (argparse) a declaration will typically look like:
+        foo=hello
+    or
+        foo="hello world"
+
+  Returns: (key,value) tuple"""
+  items = s.split('=')
+  k = items[0].strip()  # Remove whitespace around keys
+
+  if len(items) <= 1:
+    raise argparse.ArgumentTypeError(
+        "Couldn't parse label '{}' into k=v format.".format(s))
+
+  v = '='.join(items[1:])
+  return (k, v)
+
+
+# ----------------------------------------------------------------------------
+def _parse_json(argname, json_string, expected_type):
```

Comment:
nice, if we've got access to python, why not pass everything via 1 arg as a json dict?

Reply:
agree